Foundation/Array/Chunked/Unboxed.hs

```diff
@@ -138,13 +144,26 @@ elem el (ChunkedUArray array) = loop 0
                 True  -> True
                 False -> loop (i+1)
 
--- | TODO: Improve implementation.
+-- | Fold a `ChunkedUArray' leftwards lazily. Implemented internally using a double
+-- fold on the nested Array structure. Other folds implemented analogously.
+foldl :: PrimType ty => (a -> ty -> a) -> a -> ChunkedUArray ty -> a
+foldl f initialAcc (ChunkedUArray cua) = A.foldl (U.foldl f) initialAcc cua
+
+foldl' :: PrimType ty => (a -> ty -> a) -> a -> ChunkedUArray ty -> a
+foldl' f initialAcc (ChunkedUArray cua) = A.foldl' (U.foldl' f) initialAcc cua
+
+foldr :: PrimType ty => (ty -> a -> a) -> a -> ChunkedUArray ty -> a
+foldr f initialAcc (ChunkedUArray cua) = A.foldr (flip $ U.foldr f) initialAcc cua
+
 minimum :: (Ord ty, PrimType ty) => C.NonEmpty (ChunkedUArray ty) -> ty
-minimum = Data.List.minimum . toList . C.getNonEmpty
+minimum cua = foldl' min (unsafeIndex cua' 0) cua'
```

Comment:
`foldl1'` for this data type is rather complex because the contained arrays may or may not be empty. I am pondering the best way to write this. Currently I just work around this by supplying the first element as accumulator and dropping it form the array.